packages/react-button/src/components/Button/useButtonState.ts

```diff
@@ -4,40 +4,56 @@ import { ButtonState } from './Button.types';
 
 /**
  * The useButton hook processes the Button draft state.
- * @param draftState - Button draft state to mutate.
+ * @param state - Button draft state to mutate.
```

Comment:
On thing here is that judging from naming, i originally was thinking `draftState` rather than simply `state` as a variable name would be a better indication that the object will be mutated. Otherwise, especially now that it returns the state, it might be interpreted to not mutate like normal circumstances.

Reply:
The comment is pretty clear that it mutates it of course, just not everyone thoroughly reads comments but they do read param names and might infer an expectation.

If there's a better term than `draftState`, maybe `partialState` or something... that is more clear, I think that would be better.

Reply:
I was using the convention that the other components have gone with. I can push back to get some better naming but that should probably not block this PR.